Readability.js

```diff
@@ -1661,10 +1661,27 @@ Readability.prototype = {
         var input = node.getElementsByTagName("input").length;
 
         var embedCount = 0;
-        var embeds = node.getElementsByTagName("embed");
+        var videoCount = 0;
+        var embeds = this._concatNodeLists(
+          node.getElementsByTagName("object"),
+          node.getElementsByTagName("embed"),
+          node.getElementsByTagName("iframe"));
+
         for (var ei = 0, il = embeds.length; ei < il; ei += 1) {
-          if (!this.REGEXPS.videos.test(embeds[ei].src))
+          var attributeValues = [].map.call(node.attributes, function(attr) {
```

Comment:
For this one, I also followed code in [`_clean`](https://github.com/mozilla/readability/blob/9b8ce0b3545a48f1a66079fc6855671d30105682/Readability.js#L1484) method as well.

I wonder which one is more expensive :

- running regex once for a long string, or
- running regex multiple times for a short string.

I'm more experienced in Go than JS, and in Go (AFAIK) running regex once is cheaper than running regex multiple times. With that said, I thought JS is the same in that aspect, which is why `_clean` method do it like this.

So, what do you think ? Should I make it loop over the attributes, or keep it like this ?

Reply:
It depends which is the common case. Here, we have a list of embeds/iframes/objects. Really, we should be checking the actual `src` attributes instead of blanket checking all the attributes and the innerHTML... but I guess we can do something like this.

However, I just realized... you're looping ove rthe `embeds` array, but you're checking `node.attributes` - and you never assign to `node`, so it's still the node ancestor in which we're checking for embeds. We should be checking `embeds[ei]` for these attributes etc.

Reply:
> you're looping over the `embeds` array, but you're checking `node.attributes`

:man_facepalming: fixed in [0a23f5](https://github.com/mozilla/readability/pull/526/commits/0a23f536326a5af02b7b10f5771cddfa82385217), now it will check the `embeds` properly.

> I'm not a fan of the joining up of the attributes like this.

I've done a small test and the difference between two regex case is negligible. So, I've decided to use the loop because IMHO it's more readable than joining attributes using map.